contracts/terraswap_factory/src/contract.rs

```diff
@@ -43,6 +47,9 @@ pub fn execute(deps: DepsMut, env: Env, info: MessageInfo, msg: ExecuteMsg) -> S
             pair_code_id,
         } => execute_update_config(deps, env, info, owner, token_code_id, pair_code_id),
         ExecuteMsg::CreatePair { asset_infos } => execute_create_pair(deps, env, info, asset_infos),
+        ExecuteMsg::AddNativeTokenDecimals { denom, decimals } => {
+            execute_add_native_token_decimals(deps, env, info, denom, decimals)
```

Comment:
what if the decimal is incorrect?

Reply:
That's why we only allow this function to the owner

Reply:
I was wondering how we can deal with it if the owner add wrong decimal by mistake. may be we can allow to update decimal in case of denom duplication..?

Reply:
have to overwrite it. I'll fix it

Reply:
https://github.com/terraswap/terraswap/pull/52/commits/dc7448703246decaf783529a0038764678e808b9